### DISCUSSION TITLE
Tests should be in Rust

### DISCUSSION THREAD

**Discussion by CRossel87a** [otter-sec/anchor#2848] [General]
Hello everyone,

Over the past week, I've been deeply immersed in Solana's Rust libraries and was thrilled to discover Anchor, which significantly streamlines the process of creating smart contracts. However, I find myself perplexed and somewhat disenchanted by the decision to integrate JavaScript into this ecosystem. The Rust frameworks at our disposal are more than capable of facilitating contract interactions and testing. Rust's purity and efficiency are among its greatest strengths, and it feels counterintuitive to dilute this with JavaScript, particularly given that production software in JS might not match Solana's speed. It's a curious choice that I'm keen to understand better.

I am now converting the tutorials to pure Rust for my own usage

**Comment by acheroncrypto** [comment]
You're free to write your tests in Rust, the next release will also support a basic Rust test template (https://github.com/coral-xyz/anchor/pull/2805).

The reason why the default tests are in JS/TS is because it's what most people use/prefer, and it also allows for a more realistic integration test experience.

**Reply by puhtaytow** [reply]
It would be cool tho if cli during initialization of a new project could ask if one prefer rust or ts setup.
